modules/sdk/gradle-plugins/src/main/java/com/liferay/gradle/plugins/tasks/WatchTask.java

```diff
@@ -402,75 +460,75 @@ private long _installBundle(
 
 		long bundleId = -1;
 
-		String url = _getReferenceInstallURL(file);
+		String url = _getReferenceInstallURL(file, getBundleSymbolicName());
 
 		String response = _sendGogoShellCommand(
 			gogoShellClient, "install " + url);
 
-		if (start) {
-			Matcher matcher = _installResponsePattern.matcher(response);
+		Matcher matcher = _installResponsePattern.matcher(response);
 
-			if (matcher.matches()) {
-				if (logger.isQuietEnabled()) {
-					logger.quiet("Installed bundle at {}", file);
-				}
+		if (matcher.matches()) {
+			if (logger.isQuietEnabled()) {
+				logger.quiet("Installed bundle at {}", file);
+			}
 
-				String bundleIdString = matcher.group(1);
+			String bundleIdString = matcher.group(1);
 
-				bundleId = Long.parseLong(bundleIdString);
+			bundleId = Long.parseLong(bundleIdString);
+		}
 
-				_startBundle(bundleId, gogoShellClient);
-			}
-			else {
-				logger.error("Unable to install bundle: {}", response);
-			}
+		if (start) {
+			_startBundle(bundleId, gogoShellClient);
+		}
+
+		if (bundleId < 0) {
+			logger.error("Unable to install bundle: {}", response);
 		}
 
 		return bundleId;
 	}
 
-	private void _installBundleFully() throws IOException {
+	private void _installOrUpdateBundle(
+			long bundleId, GogoShellClient gogoShellClient)
+		throws IOException {
+
 		File bundleDir = getBundleDir();
 
-		File manifestFile = new File(bundleDir, "META-INF/MANIFEST.MF");
+		if (bundleId > 0) {
+			_updateBundle(bundleDir, bundleId, gogoShellClient);
+		}
+		else {
+			bundleId = _installBundle(bundleDir, gogoShellClient, true);
+		}
 
-		try (GogoShellClient gogoShellClient = new GogoShellClient();
-			InputStream inputStream = new FileInputStream(manifestFile)) {
+		_installedAttributes.put(bundleDir, _getAttributes(bundleDir));
 
-			Manifest manifest = new Manifest(inputStream);
+		FileCollection fileCollection = getFragments();
 
-			Attributes attributes = manifest.getMainAttributes();
+		boolean installedFragment = false;
 
-			String bundleSymbolicName = attributes.getValue(
-				"Bundle-SymbolicName");
+		if (fileCollection != null) {
+			Set<File> files = fileCollection.getFiles();
 
-			long bundleId = _getBundleId(bundleSymbolicName, gogoShellClient);
+			for (File file : files) {
+				long fragmentBundleId = _installBundle(
+					file, gogoShellClient, false);
 
-			if (bundleId > 0) {
-				_updateBundle(bundleDir, bundleId, gogoShellClient);
-			}
-			else {
-				bundleId = _installBundle(bundleDir, gogoShellClient, true);
+				if (fragmentBundleId > 0) {
+					installedFragment = true;
+				}
 			}
+		}
 
-			File outputFile = getOutputFile();
-
-			String bundleIdString = String.valueOf(bundleId);
-
-			Files.write(
-				outputFile.toPath(),
-				bundleIdString.getBytes(StandardCharsets.UTF_8));
-
-			_installedAttributes.put(bundleDir, attributes);
-
-			FileCollection fileCollection = getFragments();
+		File outputFile = getOutputFile();
```

Comment:
The newline below isn't needed since `bundleIdString` does not depend on the `outputFile`.  These variables can also be sorted alphabetically